config/types/luks.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2016 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package types
+
+import (
+	"errors"
+
+	"github.com/coreos/ignition/config/validate/report"
+)
+
+var (
+	ErrLuksNoDevice     = errors.New("no device provided for encrypted volume setup")
+	ErrLuksKeyAndSource = errors.New("luks volume has both key and keysource defined")
+)
+
+type Luks struct {
+	Name      string `json:"name,omitempty"`
+	Device    Path   `json:"device,omitempty"`
+	Key       string `json:"key,omitempty"`
```

Comment:
These two should probably be collapsed into one key. Maybe use a URL and have immediate keys represented as a Data URL.